restic/restore.go

```diff
@@ -95,20 +95,30 @@ func (r *restoreStats) ToJson() []byte {
 }
 
 // Restore takes a snapshotID and a method to create a restore job.
-func (r *RestoreStruct) Restore(snapshotID, method string, snaps []Snapshot, restoreDir, restoreFilter string, verifyRestore bool) {
+func (r *RestoreStruct) Restore(snapshotID, method string, snaps []Snapshot, restoreDir, restoreFilter string, verifyRestore bool, tags []string) {
 
 	r.setState(method, restoreDir, restoreFilter, verifyRestore)
 
-	r.errorMessage = r.restoreCommand(snapshotID, method, snaps)
+	r.errorMessage = r.restoreCommand(snapshotID, method, snaps, tags)
 }
 
-func (r *RestoreStruct) restoreCommand(snapshotID, method string, snaps []Snapshot) error {
+func (r *RestoreStruct) restoreCommand(snapshotID, method string, snaps []Snapshot, tags []string) error {
 	fmt.Println("Starting restore...")
 
 	snapshot := Snapshot{}
 
 	if snapshotID == "" {
-		fmt.Println("No snapshot defined, using latest one.")
+		if len(tags) > 0 {
+			tagList := []string{}
+			for i := range tags {
```

Comment:
I guess you could save the loop if you just pass in the tags, without a call to `BuildArgs()` first?

Reply:
That was the idea behind this. But you can't reference a type from the main package without creating a dependency loop unfortunately. If you have a better idea to solve this cleanly I'm all ears.

Reply:
You could move the type to its own package? Or just don't call `BuildArgs()` for this function